### ISSUE TITLE
What will this GnuCOBOL mode be about?

### ISSUE THREAD

**Issue by GitMensch** [wmealing/gnucobol-mode#1]
Note 1: I was routed here when searching for GnuCOBOL on duckduckgo (for whatever reason this repo is a top scorer...)
Note 2: there's a spelling issue, it is indeed GnuCOBOL

Side question if you don't mind: what are you personally using GnuCOBOL for?

**Comment by wmealing** [comment]
> Note 1: I was routed here when searching for GnuCOBOL on duckduckgo (for whatever reason this repo is a top scorer...)

Thats weird, I just made this not long ago, Im not trying to game the system.  Are you asking me to delete the repo ?

> Note 2: there's a spelling issue, it is indeed GnuCOBOL

Emacs modes are usually in the format of 'language - mode', I do have something working based on 'cobol-mode' because its indentation behaviour is weird it doesn't work as one would expect.

> Side question if you don't mind: what are you personally using GnuCOBOL for?

I had a few things for which I want to write in GnuCOBOL, which I assumed that it would suit, but I'm not having much luck. 

1) I wanted to have a working COBOL implementation in lunatic with the additional message passing between them. I have this partially working but making this repo public might be more 'support' than its worth.    I've adopted cobweb ( https://github.com/cloudflare/cobweb not the older cobweb)  into lunatic  ( https://lunatic.solutions/ ) for fun and use that way.  I do have this working in basic use-case, but its only small part of the project that i need.
 
2) I have recently tried to call C from GnuCOBOL for tdlib (telegram) as an example, however, I'm not getting much luck.  It might be my incompetence or it might be tdlib, Other people seem to have written bots in C, so I have to assume it is my incompetence.

3) I wanted to wrap xterm.js around COBOL to make TUI's sane for the web.   This makes a connection via a websocket, which I kinda have working.  I've had to do stupid amounts of trickery parsing JSON back and forward, it's not been a good time.

However, the significant discouragements in my attempts to call tdblib (via c) not working (probably my fault) along with JSON parsing, and different COBOL documentation (not GnuCOBOL specifically)  really complicated my learning process.

Anyways, if you want to make gnucobol-mode , I'm not attached to the outcome.

**Comment by wmealing** [comment]
TLDR:  This was where i was going to make a GnuCOBOL major mode for emacs, thats all , enjoy the story.

**Comment by wmealing** [comment]
I did not know about generate json, that may have saved me some time if I had known.

If you want to see the start of the cobol-mode check out the repos README.md, I have updated it recently.
